terraform/logical/bi.tf

```diff
@@ -41,33 +41,186 @@ resource "kubernetes_secret" "grafana_config" {
-    GF_SERVER_SERVE_FROM_SUBPATH = true
-    GF_USERS_DEFAULT_THEME       = "light"
+    GF_SERVER_ROOT_URL                      = local.grafana_url
+    GF_SERVER_SERVE_FROM_SUBPATH            = true
+    GF_USERS_DEFAULT_THEME                  = "light"
+    GF_DATABASE_TYPE                        = "mysql"
+    GF_DATABASE_HOST                        = local.db_master_host
+    GF_DATABASE_USER                        = local.db_master_username
+    GF_DATABASE_PASSWORD                    = local.db_master_password
+    GF_ANALYTICS_REPORTING_ENABLED          = false
+    GF_ANALYTICS_CHECK_FOR_UPDATES          = false
+    GF_METRICS_ENABLED                      = false
+    GF_SECURITY_COOKIE_SECURE               = true
+    GF_SECURITY_DATA_SOURCE_PROXY_WHITELIST = "1.1.1.1:1" #Disable all data source proxying
+    GF_SECURITY_COOKIE_SAMESITE             = "strict"
+    GF_SECURITY_X_XSS_PROTECTION            = true
   }
 }
 
+resource "kubernetes_config_map" "grafana_create_db_script" {
+  count = var.enable_bi ? 1 : 0
+  metadata {
+    name      = "grafana-create-db-script"
+    namespace = local.k8s_namespace_name
+  }
+
+  data = {
+    "grafana-db.sql" = file("static/grafana-db.sql")
+  }
+}
+
+
+resource "kubernetes_secret" "grafana_mysql_credentials" {
+  count = var.enable_bi ? 1 : 0
+
+  metadata {
+    name      = "grafana-mysql-credentials"
+    namespace = local.k8s_namespace_name
+  }
+  type = "Opaque"
+
+  data = {
+    host     = local.db_master_host
+    user     = local.db_master_username
+    password = local.db_master_password
+  }
+}
+
+resource "kubernetes_job" "grafana_db_create" {
+  count = var.enable_bi ? 1 : 0
+
+  metadata {
+    name      = "grafana-db-create"
+    namespace = kubernetes_namespace.kots_app.metadata[0].name
+  }
+  spec {
+    template {
+      metadata {}
+      spec {
+        container {
+          name  = "grafana-db-create"
+          image = "imega/mysql-client"
+          env {
+            name = "MYSQL_HOST"
+            value_from {
+              secret_key_ref {
+                name = kubernetes_secret.grafana_mysql_credentials[0].metadata[0].name
+                key  = "host"
+              }
+            }
+          }
+          env {
+            name = "MYSQL_USER"
+            value_from {
+              secret_key_ref {
+                name = kubernetes_secret.grafana_mysql_credentials[0].metadata[0].name
+                key  = "user"
+              }
+            }
+          }
+          env {
+            name = "MYSQL_PASSWORD"
+            value_from {
+              secret_key_ref {
+                name = kubernetes_secret.grafana_mysql_credentials[0].metadata[0].name
+                key  = "password"
+              }
+            }
+          }
+          command = [
+            "sh",
+            "-c",
+            "mysql --host=$MYSQL_HOST --user=$MYSQL_USER --password=$MYSQL_PASSWORD --execute=\"$(cat /scripts/grafana-db.sql)\""
```

Comment:
The current state of `grafana-db.sql` checks out for use in an `--execute` statement. There could be a future where quoting could create a problem then we could maybe use ` < /scripts/grafana-db.sql` instead? No need to edit it here, just observing a potential land mine in the far future.

Reply:
Good call. That line was generated by ChatGPT and I never really liked the cat script to execute but never went back and changed it.